lib/agoneum/web/views/layout_view.ex

```diff
@@ -1,3 +1,23 @@
 defmodule Agoneum.Web.LayoutView do
   use Agoneum.Web, :view
+
+  alias Agoneum.Account.Session
+
+  def current_user_name(conn) do
+    user = Session.current_user(conn)
+    user_display_name(user)
+  end
+  defp user_display_name(user) when not is_nil(user) do
+    user_display_name(user.email, user.name)
+  end
+  defp user_display_name(_) do
+    "Guest"
+  end
+  defp user_display_name(_, name) when not is_nil(name) do
+    name
+  end
```

Comment:
I think you could avoid the guard clause by having `user_display_name(email, nil)` above `user_display_name(_, name)`, but I think that's honestly a matter of preference.